crates/mun_codegen/src/ir/file_group.rs

```diff
@@ -75,14 +73,18 @@ pub(crate) fn ir_query(db: &impl IrDatabase, file_id: hir::FileId) -> Arc<FileGr
 
     let dispatch_table = dispatch_table_builder.build();
 
-    let abi_types = gen_abi_types(&db.context());
-    let mut type_table_builder = TypeTableBuilder::new(
-        db,
-        &llvm_module,
-        &abi_types,
-        intrinsics_map.keys(),
-        &dispatch_table,
-    );
+    let type_context = IrTypeContext {
+        context: &db.context(),
+        target_data: &db.target_data(),
+        struct_types: Default::default(),
```

Comment:
Was there a particular reason that you chose not to create a "helper" construction function? Is there a case where you want to start with some `struct_types` for example?

Reply:
Its because `IrTypeContext::new(&db.context(),..)` doesn't work because the value of `db.context()` is dropped immediately in the call. So I would have to store the context first (and the target_data) which made the code longer and less clear IMHO.